qcodes/instrument/base.py

```diff
@@ -401,6 +424,259 @@ def validate_status(self, verbose: bool = False) -> None:
+
+        .. literalinclude:: ../../../qcodes/instrument_drivers/new_style.py
+            :pyobject: InstrumentWithInitValue
+
+        Which will be instantiated as
+
+        .. nbinput:: ipython3
+            :execution-count: 1
+
+            from qcodes.instrument_drivers.new_style import InstrumentWithInitValue
+            instr = InstrumentWithInitValue(name="my_instr", some_arg=123)
+            instr.print_readable_snapshot(update=True)
+
+        .. code-block:: python
+
+            my_instr:
+                parameter value
+            ----------------------------------------------------------------------------
+            IDN  :
+                {'vendor': None, 'model': 'my_instr', 'serial': None, 'firmware': None}
+            time :  123 (s)
+
+        **Inheritance and overriding parameters:**
+
+        .. literalinclude:: ../../../qcodes/instrument_drivers/new_style.py
+            :pyobject: MyInstrumentDriver
+
+        .. nbinput:: ipython3
+            :execution-count: 2
+
+            from qcodes.instrument_drivers.new_style import MyInstrumentDriver
+
+            instr = MyInstrumentDriver(name="instr", init_freq=8)
+            instr.freq(10)
+            instr.print_readable_snapshot(update=True)
+            print("\ninstr.time.label: ", instr.time.label)
+
+        .. code-block:: python
+
+            instr:
+                parameter value
+            ----------------------------------------------------------------------------
+            IDN  :
+                {'vendor': None, 'model': 'instr', 'serial': None, 'firmware': None}
+            freq :  10 (Hz)
+            time :  3 (s)
+
+            instr.time.label:  Time
+
+        .. literalinclude:: ../../../qcodes/instrument_drivers/new_style.py
+            :pyobject: SubMyInstrumentDriver
+
+        .. nbinput:: ipython3
+            :execution-count: 3
+
+            from qcodes.instrument_drivers.new_style import SubMyInstrumentDriver
+
+            sub_instr = SubMyInstrumentDriver(name="sub_instr", init_freq=99)
+            sub_instr.time(sub_instr.time() * 2)
+            sub_instr.print_readable_snapshot(update=True)
+            print("\nsub_instr.time.label: ", sub_instr.time.label)
+
+        .. code-block:: python
+
+            sub_instr:
+                parameter value
+            ----------------------------------------------------------------------------
+            IDN       :
+                {'vendor': None, 'model': 'sub_instr', 'serial': None, 'firmware'...
+            amplitude : 0 (V)
+            freq      : 99 (Hz)
+            time      : 14 (s)
+
+            sub_instr.time.label:  Time long
+    """
+    if not method.__name__.startswith(_DECORATED_METHOD_PREFIX):
+        raise ValueError(
+            f"Only methods prefixed with {_DECORATED_METHOD_PREFIX!r} can be decorated "
+            f"with `add_parameter` decorator."
+        )
+
+    @wraps(method)  # preserves info like `__doc__` and signature
```

Comment:
excessive comment :)
```suggestion
    @wraps(method)
```